analysis/basestats/analyze.h

```diff
@@ -2,12 +2,19 @@
 
 #include "request.h"
 #include <iostream>
+#include <map>
 #include <unordered_map>
 #include <unordered_set>
+#include <vector>
 
 class Analysis {
 protected:
     std::unordered_map<std::string, int64_t > counters;
+    std::unordered_map<std::string, int64_t > ranks;
+    std::map<int64_t, int64_t > reqranks;
+    // tuple: oid, last_seen_ts, <2 days boolean
+    std::unordered_map<std::string, std::tuple<int, double, bool> > reqs;
```

Comment:
Would be easier to read and program to use a struct

struct requestStat{
 std::string oid;
 double last_seen_ts;
 bool 2_days_see;
};

Reply:
Also, there seems to be some duplication and type mismatch: reqs holds the oid already in the key (as string) and then in the tuple as int (which won't work).

Reply:
After integrating with ranks, the type should be
struct requestStat{
uint64_t request_count;
double last_seen_ts;
bool 2_days_see;
};
std::unordered_map<std::string, requestStat > reqs;

and then later you can simply do
reqs[req.oid].request_count++;
etc.